### DISCUSSION TITLE
Weblate improvements @ R Project Sprint 2023

### DISCUSSION THREAD

**Discussion by daroczig** [r-devel/r-project-sprint#10] [Ideas]
Planning to work on the [already listed ideas](https://github.com/orgs/r-devel/projects/2) and collaborate with others to improve the current service at https://translate.rx.studio

**Comment by daroczig** [comment]
cc @lawremi @MichaelChirico if you might be interested

**Comment by hturner** [comment]
Thanks @daroczig. It would be great if you could make a PR for a proposal on this, with some background for people new to the Weblate server project.

@alswajiab this project may be of interest to you with your experience in sys admin and translation of technical documentation.

**Reply by daroczig** [reply]
I've written up a draft and published at https://contributor.r-project.org/r-project-sprint-2023/projects/weblate-improvements/ (and at https://github.com/r-devel/r-project-sprint-2023/discussions/16 as per the instructions) -- any feedback is highly appreciated.

**Reply by hturner** [reply]
Looks great, thanks!

**Comment by jeroen** [comment]
This is maybe a tangent, but are there efforts on the way to support (partial) translation of R manual pages? rOpenSci is interested in translating documentation of packages. Does R already support this, or should I start a separate discussion for this?

**Comment by aitap** [comment]
Not as far as I know. It's possible to stick something together by
combining \Sexpr[stage=render]{} with gettext(), but the resulting user
experience for both the documentation writer and the translator is
sub-optimal.

Additionally, PDF documentation becomes very fun once you need to
typeset non-Latin scripts. It's not impossible, but will likely mean
requiring a Unicode TeX engine like XeLaTeX and adding some facilities
for the package to declare a font with which its documentation can be
built: https://stat.ethz.ch/pipermail/r-package-devel/2023q2/009197.html

For now, a package can declare its Language: in the DESCRIPTION, but
that's it.

**Comment by eliocamp** [comment]
It'd be interesting to talk to r core about it. We've exchanged emails with Hadley about this some years ago and it lead to this proof of concept: https://github.com/eliocamp/translated

It uses some weird feature of `help` that allows you to type `es?function` to get to an .Rd with a `function-es` alias. I don't think this is the indented use, but it's what we currently have. But before running with it it might be good to talk with r core.

A better option would be to have some mechanism for `help` to read the locale option and try to look up the appropriate documentation page, though.

**Reply by jeroen** [reply]
OK let's start a separate topic about this, to see what it will take to get proper support for this in R: https://github.com/r-devel/r-project-sprint-2023/discussions/19
